homeassistant/components/switch/hydrawise.py

```diff
@@ -0,0 +1,112 @@
+"""
+Support for Hydrawise cloud.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.hydrawise
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.hydrawise import (
+    ALLOWED_WATERING_TIME, CONF_ATTRIBUTION, CONF_WATERING_TIME,
+    DATA_HYDRAWISE, DEFAULT_WATERING_TIME, HydrawiseEntity, SWITCHES)
+from homeassistant.components.switch import PLATFORM_SCHEMA, SwitchDevice
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, CONF_MONITORED_CONDITIONS)
+
+DEPENDENCIES = ['hydrawise']
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=list(SWITCHES)):
+        vol.All(cv.ensure_list, [vol.In(SWITCHES)]),
+    vol.Optional(CONF_WATERING_TIME, default=DEFAULT_WATERING_TIME):
+        vol.All(vol.In(ALLOWED_WATERING_TIME)),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up a sensor for a hydrawise device."""
+    hydrawise = hass.data[DATA_HYDRAWISE].data
+
+    default_watering_timer = config.get(CONF_WATERING_TIME)
+
+    sensors = []
+    for sensor_type in config.get(CONF_MONITORED_CONDITIONS):
+        # create a switch for each zone
+        for zone in hydrawise.relays:
+            sensors.append(
+                HydrawiseSwitch(default_watering_timer,
+                                zone,
+                                sensor_type))
+
+    add_devices(sensors, True)
+    return True
```

Comment:
Not needed.

Reply:
I believe it's needed because if I remove it the initial state after starting HA is always off and doesn't match the actual switch state until an update occurs (many seconds later).

Reply:
Then you removed the wrong `True`. You should remove the return statement that returns `True`, not the `True` in the `add_devices` call.

Reply:
Got it. Thanks! Makes sense now.